Foundation/testsuite/src/PathTest.cpp

```diff
@@ -1619,6 +1619,27 @@ void PathTest::testWindowsSystem()
 #endif
 }
 
+void PathTest::testSelf()
+{
+	std::string self = Path::self();
+	std::cout << self << std::endl;
+
+#if POCO_OS == POCO_OS_MAC_OS_X                               \
+    || POCO_OS == POCO_OS_FREE_BSD                            \
+	|| POCO_OS == POCO_OS_SOLARIS                             \
+	|| POCO_OS == POCO_OS_LINUX                               \
+	|| POCO_OS == POCO_OS_ANDROID
+
+	assertTrue(!self.empty());
+	Path p(self);
+
+	assertTrue(p.isAbsolute());
+	assertTrue(p.isFile());
+	assertTrue(self.find("testrunner") != std::string::npos);
+
```

Comment:
```c+=
#else
   std::cout << "Not implemented";
```

Reply:
Added tests for Windows and message for platforms where functionality is not implemented.